lib/routes/51read/article.ts

```diff
@@ -0,0 +1,90 @@
+import { load } from 'cheerio';
+import cache from '@/utils/cache';
+import ofetch from '@/utils/ofetch';
+import type { Route, DataItem } from '@/types';
+
+export const route: Route = {
+    path: '/article/:id',
+    name: '章节',
+    url: 'm.51read.org',
+    maintainers: ['lazwa34'],
+    example: '/51read/article/152685',
+    parameters: { id: '小说 id, 可在对应小说页 URL 中找到' },
+    categories: ['reading'],
+    features: {
+        requireConfig: false,
+        requirePuppeteer: false,
+        antiCrawler: false,
+        supportBT: false,
+        supportPodcast: false,
+        supportScihub: false,
+    },
+    radar: [
+        {
+            source: ['m.51read.org/xiaoshuo/:id'],
+            target: '/article/:id',
+        },
+        {
+            source: ['51read.org/xiaoshuo/:id'],
+            target: '/article/:id',
+        },
+    ],
+    handler,
+};
+
+async function handler(ctx) {
+    const { id } = ctx.req.param();
+    const link = `https://m.51read.org/xiaoshuo/${id}`;
+    const $book = load(await get(link));
+
+    const chapter = `https://m.51read.org/zhangjiemulu/${id}`;
+    const $chapter = load(await get(chapter));
+
+    const pageLength = $chapter('.ml-page select')
+        .find('option')
+        .map((_, option) => option.attribs.value)
+        .toArray().length;
+
+    const item: DataItem[] = [];
+    if (pageLength > 1) {
+        item.push(...(await createItem(chapter, pageLength - 1)));
+    }
+    item.push(...(await createItem(chapter, pageLength)));
```

Comment:
Don't do page turns. Providing content from the last page is enough

Reply:
It has been modified.